src/rule.jl

```diff
@@ -1151,13 +1151,25 @@ function Base.showerror(io::IO, error::RuleMethodError)
         arguments_spec = join(spec, ", ")
         meta_spec      = rule_method_error_extract_meta(error.meta)
 
+        # Print list of existing rules for this node
+        println(io, "\n\nExisting rule(s) for node:\n")
+        all_rules = methods(ReactiveMP.rule)
+        this_node_rules = all_rules[get_node_from_rule_method.(all_rules) .== "$(error.fform)"]
+        for node_rule in this_node_rules
+            node_name = get_node_from_rule_method(node_rule)
+            node_inputs = get_messages_from_rule_method(node_rule)
+            if typeof(node_inputs) !== Vector{Any}
```

Comment:
Why do we need this check?

Reply:
`get_messages_from_rule_method(node_rule)` will return `Any[]` on an empty list (see rule.jl#1297).

For example:
```julia
> all_rules = methods(ReactiveMP.rule)
> this_node_rules = all_rules[ReactiveMP.get_node_from_rule_method.(all_rules) .== "MvNormalMeanVariance"]
> ReactiveMP.get_messages_from_rule_method(this_node_rules[1])
```

Without the type check, this snippet would print a rule for an `Any` type (which doesn't exist).

Reply:
Ah, perhaps then we should use `!isempty(node_inputs)`?